NewRelic/NewRelicInteractorInterface.php

```diff
@@ -14,107 +16,147 @@
+     *
+     * {@link https://docs.newrelic.com/docs/agents/php-agent/php-agent-api/newrelic_ignore_transaction}
      */
-    public function ignoreTransaction();
+    public function ignoreTransaction(): void;
 
     /**
-     * @param string $name
-     * @param array  $attributes
+     * Record a custom event with the given name and attributes.
      *
-     * @return void
+     * {@link https://docs.newrelic.com/docs/agents/php-agent/php-agent-api/newrelic_record_custom_event}
      */
-    public function addCustomEvent($name, array $attributes);
+    public function addCustomEvent(string $name, array $attributes): void;
 
     /**
-     * @param string $name
-     * @param string $value
+     * Add a custom metric (in milliseconds) to time a component of your app not captured by default.
      *
-     * @return void
+     * {@link https://docs.newrelic.com/docs/agents/php-agent/php-agent-api/newreliccustommetric-php-agent-api}
      */
-    function addCustomMetric($name, $value);
+    public function addCustomMetric(string $name, float $value): bool;
 
     /**
-     * @param string $name
-     * @param string $value
+     * {@link https://docs.newrelic.com/docs/agents/php-agent/php-agent-api/newrelic_add_custom_parameter}.
      *
-     * @return void
+     * @param string|integer|float $value should be a scalar
      */
-    function addCustomParameter($name, $value);
+    public function addCustomParameter(string $name, $value): bool;
 
     /**
-     * @return string
+     * Returns a New Relic Browser snippet to inject in the head of your HTML output.
+     *
+     * {@link https://docs.newrelic.com/docs/agents/php-agent/php-agent-api/newrelic_get_browser_timing_header}
      */
-    function getBrowserTimingHeader();
+    public function getBrowserTimingHeader(bool $includeTags = true): string;
 
     /**
-     * @return string
+     * Returns a New Relic Browser snippet to inject at the end of the HTML output.
+     *
+     * {@link https://docs.newrelic.com/docs/agents/php-agent/php-agent-api/newrelic_get_browser_timing_footer}
      */
-    function getBrowserTimingFooter();
+    public function getBrowserTimingFooter(bool $includeTags = true): string;
 
     /**
-     * @return void
+     * Disable automatic injection of the New Relic Browser snippet on particular pages.
+     *
+     * {@link https://docs.newrelic.com/docs/agents/php-agent/php-agent-api/newrelic_disable_autorum}
      */
-    function disableAutoRUM();
+    public function disableAutoRUM(): bool;
 
     /**
-     * @param string $msg
+     * Use these calls to collect errors that the PHP agent does not collect automatically and to set the callback for
+     * your own error and exception handler.
      *
-     * @return void
+     * {@link https://docs.newrelic.com/docs/agents/php-agent/php-agent-api/newrelic_notice_error}
      */
-    function noticeError($msg);
+    public function noticeThrowable(\Throwable $e, string $message = null): void;
```

Comment:
`newrelic_notice_error` allow string or exception. But IMHO `noticeThrowable` should only accept a Throwable. allowing both string and exception is the same method add complexity.

I suggest to expose a dedicated method `noticeErrorMessage` or promote usage of `noticeThrowable(new \Exception($message))`

Reply:
I was really hesitating with this. 
As you see here: https://docs.newrelic.com/docs/agents/php-agent/php-agent-api/newrelic_notice_error

We have the following signatures: 

```php
newrelic_notice_error(string $message[, Throwable|Exception $e])
newrelic_notice_error(Throwable|Exception $e)
newrelic_notice_error(int $errno, string $errstr[, string $errfile[, int $errline[, string $errcontext]]])
```

Just doing `newrelic_notice_error(string $message)` does not really make any sense. Then you could just pack your string in an `\Exception`. But I thought that it may be a use case where you had an exception but wanted to use a different `$message`.